### ISSUE TITLE
Type definitions for DCS

### ISSUE THREAD

**Issue by sanketshevkar** [accordproject/concerto#943]
# Bug Report 🐛

Type definitions for a DCS model are missing. Add generated typescript types of DCS Concerto Model to concerto-types module.

## Expected Behavior

Type of DCS object should be centrally defined in concerto-types.

## Possible Solution

We'll require a code generation script similar to [this](https://github.com/accordproject/concerto/blob/main/packages/concerto-types/scripts/codegen.js) that will convert the [DCS concerto model](https://github.com/accordproject/concerto/blob/3dac62f4690a81fa1754b9d9116cba869e22ac3d/packages/concerto-core/lib/decoratormanager.js#L37) to its corresponding typescript definitions.

**Comment by Ayush1404** [comment]
Hi @sanketshevkar,

I hope you're doing well! I'm interested in contributing, but I'm having a bit of trouble understanding the issue. Could you kindly provide a bit more detail or clarify it for me?

Thanks so much in advance!

**Comment by sanketshevkar** [comment]
Hi @Ayush1404 
I've updated the issue description with some reference link.

**Comment by sanketshevkar** [comment]
@Ayush1404 just a heads up, I might pick this up next week on Wednesday, if not already started. Please let me know if you'd already started work on this.

**Comment by Ayush1404** [comment]
hey @sanketshevkar , sorry for the late reply , was a little busy due to college exams . We can do two things we can make changes in existing code-gen script to include DCS_MODEL from the decoratormanager.js file like this 
<img width="1215" alt="Screenshot 2024-12-01 at 5 18 05 PM" src="https://github.com/user-attachments/assets/024691f3-30c2-4c2a-bf83-cb2543b0a7c5">
or we can generate a new script , something like this 
<img width="1199" alt="Screenshot 2024-12-01 at 5 21 15 PM" src="https://github.com/user-attachments/assets/e6d13cea-3a2e-4582-9f0c-31ae941ab17a">
what do you think , which one do you think should we go for ?

**Comment by sanketshevkar** [comment]
Thanks @Ayush1404. I'm sorry I don't want to bother you during your exams, hope you are done with them.

The first approach seems fit to to me.

**Comment by Ayush1404** [comment]
@sanketshevkar No problem , the exams are over now . Do i have to make any changes in it , before making a PR ?

**Comment by sanketshevkar** [comment]
Idts, maybe just raise the PR following our contributors guideline and do not forget the DCO-Signoff :)

**Comment by sanketshevkar** [comment]
@Ayush1404 are you still working on this.

**Comment by fuyalasmit** [comment]
Hi @sanketshevkar, I noticed this issue is in the backlog
I'd be happy to work on this, would you assign this to me ?

**Comment by fuyalasmit** [comment]
I saw the codebase meanwhile, and I have thought of a roadmap to solve the issue:

1. Load the DCS_MODEL from decoratormanager.js to modelManager
2. Then, process the model using TypescriptVisitor
3. Writing into correct output directory

**Comment by TC5022** [comment]
Hi @fuyalasmit , are you still working on this issue?

**Comment by fuyalasmit** [comment]
> Hi [@fuyalasmit](https://github.com/fuyalasmit) , are you still working on this issue?

Hi, I suggested a approach in brief before, but I was involved in other issues in the meantime.
However, I'm planning to look at this one again soon. If you're interested, we could collaborate on solving it together. Let me know if that works for you!

**Comment by fuyalasmit** [comment]
> Hi [@fuyalasmit](https://github.com/fuyalasmit) , are you still working on this issue?

> Hi [@fuyalasmit](https://github.com/fuyalasmit) , are you still working on this issue?

Hi, I suggested a approach in brief before, but I was involved in other issues in the meantime.
However, I'm planning to look at this one again soon. If you're interested, we could collaborate on solving it together. Let me know if that works for you.

**Comment by fuyalasmit** [comment]
> Hi [@fuyalasmit](https://github.com/fuyalasmit) , are you still working on this issue?

Hi, I suggested a approach in brief before, but I was involved in other issues in the meantime.
However, I'm planning to look at this one again soon. If you're interested, we could collaborate on solving it together. Let me know if that works for you!

**Comment by TC5022** [comment]
> > Hi [@fuyalasmit](https://github.com/fuyalasmit) , are you still working on this issue?
> 
> Hi, I suggested a approach in brief before, but I was involved in other issues in the meantime. However, I'm planning to look at this one again soon. If you're interested, we could collaborate on solving it together. Let me know if that works for you!

Sure, let's work it together. Have you made any changes in the code already?

**Comment by fuyalasmit** [comment]
Haven't made any changes regarding this issue, I was waiting for the maintainer to respond since he stated 

> a new PR is to be opened in concerto-metamodel repo [here](https://github.com/accordproject/concerto/pull/952)

By the way, have you thought of any approach to tackling this?

**Comment by fuyalasmit** [comment]
Okay, if you are starting, make sure you clone [this](https://github.com/fuyalasmit/concerto) and work here, so that it will be easier for me too, since I already have it set up locally. I couldn’t see your fork, so collaborating this way will help.
Also, make sure you use co-authored commits.

**Comment by TC5022** [comment]
> Okay, if you are starting, make sure you clone [this](https://github.com/fuyalasmit/concerto) and work here, so that it will be easier for me too, since I already have it set up locally. I couldn’t see your fork, so collaborating this way will help. Also, make sure you use co-authored commits.

We will first have to define the DCS model in [this](https://github.com/TC5022/concerto-metamodel) repository (linked my fork) instead of concerto and export it from there. After that we will have to import it [here](https://github.com/TC5022/concerto) so that the overall exports work. 

@sanketshevkar, I wanted to clarify where should I define the DCS model in the concerto-metamodel repo. Will defining it as a variable in the [utils file](https://github.com/sanketshevkar/concerto-metamodel/blob/main/lib/metamodelutil.js) work?

**Comment by sanketshevkar** [comment]
Apologies I was caught up in some work. I'm assuming @Ayush1404 is no longer working on this issue. But I'll still wait for his response.

**Comment by sanketshevkar** [comment]
@TC5022 I've requested changes on your PR. Once those are in. Can @fuyalasmit or you just clone @Ayush1404 's changes and update import there?

With this we can have your both as well as @Ayush1404 's changes in the codebase. I don't want anyone's work get wasted here. Please avoid doing a **FORCE PUSH** to the PR we'll loose the commits.

**Comment by fuyalasmit** [comment]
yeah sure

**Comment by fuyalasmit** [comment]
yeah sure, @TC5022 will do the needed work in concerto-metamodel, when it gets done there, I'll fix the changes for type definition for DCS
is that okay ?

**Comment by fuyalasmit** [comment]
hey, what's the progress so far ?
